src/transformState.js

```diff
@@ -5,7 +5,30 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        for (const clear in state) {
+          delete state[clear];
+        }
+        break;
+
+      case 'addProperties':
+        for (const add in action) {
+          if (add === 'extraData') {
+            Object.assign(state, action[add]);
+          }
+        }
```

Comment:
Name for variables should be noun, now you have `add` it's action. Rename it please
```suggestion
        for (const add in action.extraData) {
          Object.assign(state, action[add]);
        }
```